src/VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleListEntryController.cs

```diff
@@ -258,10 +228,54 @@ public async Task<ActionResult> Move([FromBody]ListEntriesMoveRequest moveReques
 
             await _categoryMover.ConfirmMoveAsync(categories);
             await _productMover.ConfirmMoveAsync(products);
-            
+
             return NoContent();
         }
 
+        /// <summary>
+        /// Bulk delete by the search criteria.
+        /// </summary>
+        /// <param name="criteria"></param>
+        [HttpPost]
+        [Route("delete")]
+        [ProducesResponseType(typeof(void), StatusCodes.Status204NoContent)]
+        public async Task<ActionResult> Delete([FromBody] CatalogListEntrySearchCriteria criteria)
+        {
+            const int deleteBatchSize = 20;
+            var authorizationResult = await _authorizationService.AuthorizeAsync(User, criteria, new CatalogAuthorizationRequirement(ModuleConstants.Security.Permissions.Delete));
+            if (!authorizationResult.Succeeded)
+            {
+                return Unauthorized();
+            }
+
+            var idsToDelete = criteria.ObjectIds?.ToList() ?? new List<string>();
+
+            if (idsToDelete.IsNullOrEmpty())
+            {
+                var listEntries = await InnerListItemsSearchAsync(criteria);
+                idsToDelete = listEntries.ListEntries
+                    .Where(x => x.Type.EqualsInvariant(ProductListEntry.TypeName) || x.Type.EqualsInvariant(CategoryListEntry.TypeName))
+                    .Select(x => x.Id)
+                    .ToList();
+            }
+
+            if (!idsToDelete.IsNullOrEmpty())
```

Comment:
I think we don't need to check this. Its never be null. If it's empty "for" ends immediately.

Reply:
Agree. Removed.